### PR TITLE
connect: add toggle to globally disable wildcard outbound network access when transparent proxy is enabled

### PR DESCRIPTION
**Pull request by rboyer**
This adds a new config entry kind "cluster" with a single special name "cluster" where this can be controlled.

TODO
- [ ] docs

### CODE REVIEW

File: agent/structs/config_entry_cluster_oss.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package structs
+
+func (e *ClusterConfigEntry) validateEnterpriseMeta() error {
```
Comment (rboyer) [inline agent/structs/config_entry_cluster_oss.go:3]: The enterprise version of this method will be lifted from the current version of `ProxyConfigEntry`

File: agent/proxycfg/snapshot.go [hunk 2]
```diff
@@ -5,8 +5,9 @@ import (
 	"fmt"
 	"sort"
 
-	"github.com/hashicorp/consul/agent/structs"
 	"github.com/mitchellh/copystructure"
+
+	"github.com/hashicorp/consul/agent/structs"
```
Comment (freddygv) [inline agent/proxycfg/snapshot.go:10]: Should this have moved?

File: agent/xds/listeners_test.go [hunk 3]
```diff
@@ -2,21 +2,23 @@ package xds
...
 	"bytes"
+	"path/filepath"
+	"sort"
+	"testing"
+	"text/template"
+	"time"
+
 	envoy_listener_v3 "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	testinf "github.com/mitchellh/go-testing-interface"
+	"github.com/stretchr/testify/require"
```
Comment (freddygv) [inline agent/xds/listeners_test.go:13]: Should these have moved?

File: command/config/write/config_write_test.go [hunk 4]
```diff
@@ -8,11 +8,12 @@ import (
 
 	"github.com/hashicorp/consul/agent/structs"
 
+	"github.com/mitchellh/cli"
```
Comment (freddygv) [inline command/config/write/config_write_test.go:11]: Should these have moved?

**Review by freddygv — APPROVED** [review]
This LGTM, thanks!

Only had a few minor comments about imports moving around.

File: agent/proxycfg/snapshot.go (hunk 2 above)
Comment (rboyer) [reply agent/proxycfg/snapshot.go:10]: Yeah I picked this tweak up from @dnephin for invoking `goimports`:
```
let g:go_fmt_options = {
    \ 'goimports': '-local github.com/hashicorp/consul',
    \ }
```

**Pull request merged**